### PR TITLE
fix: add ability to use custom flatlist implementation

### PR DESCRIPTION
**Pull request by arpitv96171**
Add ability to use custom Flatlist as component with example

Before: Flatlist does not scroll when used inside bottomsheet

https://github.com/user-attachments/assets/780cc5b0-069a-4dfd-9a86-92ac0c654774

After: Using BottomSheetFlatlist which scrolls when used inside [@gorhom/bottom-sheet](https://ui.gorhom.dev/components/bottom-sheet)

https://github.com/user-attachments/assets/3e0af8e9-fd53-4db1-8628-d424288f9b1b

### CODE REVIEW

File: example/src/Examples/Advanced/editor-web/build/editorHtml.ts
Comment (GuySerfaty) [inline example/src/Examples/Advanced/editor-web/build/editorHtml.ts:1]: let's revert this? no need to change the bundle of the advanced example editor

File: example/src/Examples/WithBottomSheet.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import BottomSheet, { BottomSheetFlatList } from '@gorhom/bottom-sheet';
```
Comment (GuySerfaty) [inline example/src/Examples/WithBottomSheet.tsx:1]: nice ! can we also add it to the doc website? with explanations code snippet and maybe a gif?

File: example/src/App.tsx [hunk 2]
```diff
@@ -77,28 +83,33 @@ const HomeScreen = ({ navigation }: NativeStackScreenProps<any, any, any>) => {
...
-        {examples.map((example) => (
-          <Stack.Screen
-            key={example.name}
-            name={example.name}
-            component={example.component}
-          />
-        ))}
-      </Stack.Navigator>
-    </NavigationContainer>
+    <GestureHandlerRootView style={homeStyles.root}>
```
Comment (GuySerfaty) [inline example/src/App.tsx:86]: why needed? (maybe a dumb question)

File: src/RichText/Toolbar/Toolbar.tsx [hunk 3]
```diff
@@ -35,6 +37,7 @@ export function Toolbar({
   editor,
   hidden = undefined,
   items = DEFAULT_TOOLBAR_ITEMS,
+  ListComponent = FlatList,
```
Comment (GuySerfaty) [inline src/RichText/Toolbar/Toolbar.tsx:40]: This is a great ability regardless the bottom-sheet, can we add this to the props table on: https://10play.github.io/10tap-editor/docs/api/Components#toolbar

**Review by GuySerfaty — CHANGES_REQUESTED** [review]
@arpitv96171 - first of all thank you very much for that contribution, this is something I was planned to do anyway!

I left some small comments and questions.
would love @17Amir17 to take a look as well

**Comment by GuySerfaty** [comment]
@arpitv96171 - commits should follow the commit message convention
https://github.com/10play/10tap-editor/blob/main/CONTRIBUTING.md#commit-message-convention

Same on the other PR both of them all of the commit should start with `fix: `
both PRs should be patch bump on the lib version

File: example/src/App.tsx (hunk 2 above)
Comment (arpitv96171) [reply example/src/App.tsx:86]: This is the primary setup requirement to ensure [react-native-gesture-handler](https://docs.swmansion.com/react-native-gesture-handler/docs/) works in the app (which is the dependency of [@gorhom/bottom-sheet](https://ui.gorhom.dev/components/bottom-sheet/))

File: example/src/Examples/WithBottomSheet.tsx (hunk 1 above)
Comment (arpitv96171) [reply example/src/Examples/WithBottomSheet.tsx:1]: added

File: src/RichText/Toolbar/Toolbar.tsx (hunk 3 above)
Comment (arpitv96171) [reply src/RichText/Toolbar/Toolbar.tsx:40]: done

File: example/src/Examples/Advanced/editor-web/build/editorHtml.ts
Comment (arpitv96171) [reply example/src/Examples/Advanced/editor-web/build/editorHtml.ts:1]: done

**Comment by arpitv96171** [comment]
new PR #173

**Pull request closed without merging**